### PR TITLE
Repro #21034: Autocomplete flicker in SQL Editor

### PR DESCRIPTION
**Pull request by ariya**
### Status
PENDING REVIEW

### What does this PR accomplish?
- Reproduces #21034

### How to test this manually?
- `yarn test-cypress-open`
- `frontend/test/metabase/scenarios/native/reproductions/21034-autocomplete-flicker.cy.spec.js`
- Unskip repro
- The test should fail until the related issue is fixed

### Additional notes:
- Once the issue is fixed, please remove the `.skip` part (unskip the test completely)
- Make sure the test is passing and
- Merge it together with the fix

### CODE REVIEW

File: frontend/test/metabase/scenarios/native/reproductions/21034-autocomplete-flicker.cy.spec.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  });
+
+  it("should not invoke API calls for autocomplete twice in a row (metabase#18148)", () => {
+    cy.get(".ace_content")
+      .should("be.visible")
+      .type("p");
+
+    // Wait until another explicit autocomplete is triggered
+    // See https://github.com/metabase/metabase/pull/20970
+    cy.wait(1000);
```
Comment (nemanjaglumac) [inline frontend/test/metabase/scenarios/native/reproductions/21034-autocomplete-flicker.cy.spec.js:22]: We shouldn't wait arbitrarily. Is this a debounce value or how did you arrive to 1000ms?

**Review by nemanjaglumac — APPROVED** [review]
🚀

File: frontend/test/metabase/scenarios/native/reproductions/21034-autocomplete-flicker.cy.spec.js (hunk 1 above)
Comment (ariya) [reply frontend/test/metabase/scenarios/native/reproductions/21034-autocomplete-flicker.cy.spec.js:22]: Long enough to be more than `AUTOCOMPLETE_DEBOUNCE_DURATION` in `frontend/src/metabase/query_builder/components/NativeQueryEditor.jsx`. Not sure how we should get that value (if anything).

Comment (nemanjaglumac) [reply frontend/test/metabase/scenarios/native/reproductions/21034-autocomplete-flicker.cy.spec.js:22]: In theory, we can export that constant and import it in this test, then add a couple hundred milliseconds.
But the comment will suffice for now.

Comment (ariya) [reply frontend/test/metabase/scenarios/native/reproductions/21034-autocomplete-flicker.cy.spec.js:23]: Good idea, I've added a comment.

**Pull request merged**